creator-node/src/services/stateMachineManager/stateReconciliation/recoverOrphanedData.jobProcessor.ts

```diff
@@ -0,0 +1,57 @@
+import type {
+  DecoratedJobParams,
+  DecoratedJobReturnValue,
+  JobsToEnqueue
+} from '../types'
+import type {
+  RecoverOrphanedDataJobParams,
+  RecoverOrphanedDataJobReturnValue
+} from './types'
+
+const { QUEUE_NAMES } = require('../stateMachineConstants')
+
+/**
+ * Processes a job to find users who have data on this node but who do not have this node in their replica set.
+ * This means their data is "orphaned,"" so the job also merges this data back into the primary of the user's replica set and then wipes it from this node.
+ *
+ * @param {Object} param job data
+ * @param {Object} param.logger the logger that can be filtered by jobName and jobId
+ */
+module.exports = async function ({
+  logger
+}: DecoratedJobParams<RecoverOrphanedDataJobParams>): Promise<
+  DecoratedJobReturnValue<RecoverOrphanedDataJobReturnValue>
+> {
+  // TODO: Clear redis state
+
+  // TODO: Query this node's db to find all users who have data on it and save them in a redis set
+  const usersOnThisNode = {}
+
+  /**
+   * TODO: Query Discovery to find all users who have this node as their primary or secondary.
+   *       This will do some kind of while loop that requests paginated responses from
+   *       `/v1/full/users/content_node/all?creator_node_endpoint=<THIS_CONTENT_NODE>` and adds
+   *       them to a redis set until reaching the end of the pagination
+   */
+  const usersWithThisNodeInReplica = {}
```

Comment:
prob not worth but a thought - since `monitor-state` already does this, is there some potential to combine work here? might just be more work, but might also benefit - idk

Reply:
I was thinking about this and talking it over with Dheeraj -- I think it's not worth trying for now because of the added complexity with another queue and managing how stale the data is. but I do think it would be a pretty nice future optimization

Reply:
agreed